docs/notebooks/flax_basics.ipynb

```diff
@@ -0,0 +1,911 @@
+{
```

Comment:
<p>Maybe we should explain <code>FrozenDict</code>  here -- basically say that because JAX is purely functional we help you "not shoot yourself in the foot" (a common theme across Flax) so that you're aware that you can't modify these dicts directly -- the wrong thing will happen.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google/flax/pull/542/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Reply:
<p>Yes, agreed, in fact I realize that at that point I asked Jonathan about it but didn't write anything here</p>

Reply:
<p>Going to add trying mutating a frozendict to show it</p>